soc/arm/st_stm32/stm32u5/soc.c

```diff
@@ -35,6 +36,10 @@ static int stm32u5_init(void)
 	LL_ICACHE_SetMode(LL_ICACHE_1WAY);
 	LL_ICACHE_Enable();
 
+	/* Enable data cache (master port write access) */
+	LL_DCACHE_SetReadBurstType(DCACHE1, LL_DCACHE_READ_BURST_INCR);
+	LL_DCACHE_Enable(DCACHE1);
```

Comment:
What are the impact on existing code when enabling DCACHE systematically? For instance use of DMA ?

Also, rather than enabling it directly here, I think we should implement a cache driver (see `drivers/cache`), in order to allow wusage of system API `sys_cache_data_foo` from drivers.

Reply:
with

> What are the impact on existing code when enabling DCACHE systematically? For instance use of DMA ? At least I think it should be enabled conditionaly
> 
> Also, rather than enabling it directly here, I think we should implement a cache driver (see `drivers/cache`), in order to allow wusage of system API `sys_cache_data_foo` from drivers.

With 
```
	sys_cache_instr_enable();
	sys_cache_data_enable();
```

both testcase tests/drivers/dma/ : PASSED